core-s3/src/test/scala/akka/stream/alpakka/s3/GeneratorsSpec.scala

```diff
@@ -1,22 +1,64 @@
 package akka.stream.alpakka.s3
 
-import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import com.typesafe.config.ConfigFactory
+import com.typesafe.config.ConfigValueFactory
 import io.aiven.guardian.kafka.s3.Generators
 import org.scalacheck.Gen
 import org.scalatest.matchers.must.Matchers
 import org.scalatest.propspec.AnyPropSpec
 import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
 
+import scala.annotation.nowarn
+
 class GeneratorsSpec extends AnyPropSpec with Matchers with ScalaCheckPropertyChecks {
+
+  def createBasicConfigFactory(virtualDotHost: Boolean): Config = {
+    @nowarn("msg=possible missing interpolator: detected an interpolated expression")
+    val baseS3SettingsConf =
+      """
+        |buffer = "memory"
+        |disk-buffer-path = ""
+        |
+        |aws {
+        |  credentials {
+        |    provider = default
+        |  }
+        |  region {
+        |    provider = default
+        |  }
+        |}
+        |access-style = virtual
+        |list-bucket-api-version = 2
+        |validate-object-key = true
+        |retry-settings {
+        |  max-retries = 3
+        |  min-backoff = 200ms
+        |  max-backoff = 10s
+        |  random-factor = 0.0
+        |}
+        |multipart-upload {
+        |  retry-settings = ${retry-settings}
+        |}
+        |sign-anonymous-requests = true
+        |""".stripMargin
+
+    val config = ConfigFactory.parseString(baseS3SettingsConf).resolve()
+    if (virtualDotHost)
+      config.withValue("access-style", ConfigValueFactory.fromAnyRef("virtual"))
+    else
+      config.withValue("access-style", ConfigValueFactory.fromAnyRef("path"))
+  }
+
```

Comment:
Why is this change needed?

Reply:
> The second part of the PR needs some context explanation. The primary goal of this PR is solving https://github.com/aiven/guardian-for-apache-kafka/issues/118 where S3 secrets are not resolved in CI tests by PR's that are created from forks. Annoyingly when secrets are not resolved this way, rather than unset'ting the environment variable you just get an empty environment variable and/or some of the env vars are missing since the others are statically defined (i.e. https://github.com/aiven/guardian-for-apache-kafka/blob/main/.github/workflows/ci.yml#L20). This causes a NullPointerThrown exception in GeneratorsSpec because the configuration is malformed.

Explained another way, currently (without this PR), because this test relies on configuration that is initialized from environment variables it fails if the secrets cannot be resolved with 

```
[info] GeneratorsSpec:
[info] - Bucket name generators generates valid bucket names according to S3Settings with virtualDotHost *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: An unexpected java.lang.NullPointerException was thrown.
[info]     Location: (GeneratorsSpec.scala:13)
[info]     Occurred when passed generated values (
[info]       arg0 = "" // 6 shrinks
[info]     )
[info] - Bucket name generators generates valid bucket names according to S3Settings without virtualDotHost *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: An unexpected java.lang.NullPointerException was thrown.
[info]     Location: (GeneratorsSpec.scala:19)
[info]     Occurred when passed generated values (
[info]       arg0 = "" // 2 shrinks
[info]     )
[info] - Bucket name generators generates valid bucket names according to S3Settings with virtualDotHost and prefix *** FAILED ***
[info]   TestFailedException was thrown during property evaluation.
[info]     Message: An unexpected java.lang.NullPointerException was thrown.
[info]     Location: (GeneratorsSpec.scala:34)
[info]     Occurred when passed generated values (
[info]       arg0 = "" // 6 shrinks
[info]     )
```

Any of the PR's created from scala steward will fail for the same reason (see https://github.com/aiven/guardian-for-apache-kafka/runs/5270935748?check_suite_focus=true as an example).

That `NullPointerException` that is thrown is due to the fact that since some of the secrets are not being resolved, only half of the configurations for S3 are being resolved which is causing `S3Settings()` to load the configuration in a broken state.

This change fixes this by just hardcoding the configuration without it being resolved from `reference.conf` or from environment variables.